demos/interactive_face_detection_demo/cpp/main.cpp

```diff
@@ -168,9 +168,9 @@ int main(int argc, char *argv[]) {
         faceDetector.submitRequest();
 
         cv::Mat next_frame = cap->read();
-
         while (frame.data) {
             timer.start("total");
+            auto startTime = std::chrono::steady_clock::now();
```

Comment:
`next_frame = cap->read();` is the place where the timer should be started. And the period should last until `metrics.update()`, but in your implementation by that moment `startTime` is overwritten by the next iteration of the loop.

Reply:
Done in https://github.com/openvinotoolkit/open_model_zoo/pull/2647/commits/b93e6b9277a4f8c818dcee06a3a4003739a70cf3

Reply:
I'll try explain the other way around.
Given `startTime` corresponds to `next_frame = cap->read();`, `startTime` can't correspond to `prev_frame` in `update()`. You need two `startTime` vars similar to `next_frame` and `prev_frame`.
Time point for `cv::Mat next_frame = cap->read();` outside of the loop isn't recorded.

Reply:
Ok, I think I get it, https://github.com/openvinotoolkit/open_model_zoo/pull/2647/commits/63dfca4f2db586c09125af8546bdd64f26212cd6

Reply:
Actually there's `cv::Mat frame = cap->read();` much earlier, you should count it as well

Reply:
Actually there's cv::Mat frame = cap->read(); much earlier, you should count it as well

Reply:
fixed